OpenSim/Simulation/Model/Geometry.h

```diff
@@ -180,8 +150,34 @@ class OSIMSIMULATION_API Geometry : public Component {
     /// Methods to support frame as a connection, implement Component interface
     void constructConnectors() override;
 
+protected:
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return the result in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const = 0;
+
 private:
-    
+    /// Compute Transform of this geometry relative to its base frame, utilizing 
+    /// passed in state. Both transform and body_id are set in the passed-in 
+    /// decorations as a side effect.
+    void setDecorativeGeometryTransform(
```

Comment:
Since Geometry is responsible for computing the transform itself using the Frames, and the computation is identical since it depends only on Frames and the state, this common functionality is implemented in the base class. Your interpretation in the second paragraph is accurate and it will not show in doxygen because the method is private but is supposed to help developers maintaining this code.